### PR TITLE
Expand docstrings for artifacts

### PR DESCRIPTION
**Pull request by discdiver**
Adds all params to the artifact docstrings and explains keys and their format

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

**Comment by discdiver** [comment]
Screenshot of 2.10.10 `create_markdown_artifact` doc string in VS Code. Without a key, the artifact doesn't appear in the Artifacts tab of the UI and the code errors if not properly formatted.

Screenshot 2023-05-23 at 6 27 38 PM

**Comment by discdiver** [comment]
Now here's the table from the deploy docs preview for this PR:

Screenshot 2023-05-23 at 7 53 08 PM

File: src/prefect/artifacts.py [hunk 1]
```diff
@@ -100,11 +108,15 @@ async def create_markdown_artifact(
     """
     Create a markdown artifact.
 
-    Args:
-        - markdown: The markdown to create.
+    Arguments:
+        markdown: The markdown to create.
+        key: A user-provided string identifier.
+          Required for the artifact to show in the Artifacts tab in the UI.
```
Comment (serinamarie) [inline src/prefect/artifacts.py:114]: I would probably call it the Artifacts page, as it will still show in the Artifacts tab of the Flow Runs page.

**Review by serinamarie — COMMENTED** [review]
Thanks for noticing this! Also, I think you accidentally pushed changes to the docs/api-ref/rest-api/index.md file to this branch

File: src/prefect/artifacts.py (hunk 1 above)
Comment (discdiver) [reply src/prefect/artifacts.py]: done

**Comment by discdiver** [comment]
"... I think you accidentally pushed changes to the docs/api-ref/rest-api/index.md file to this branch" 

Thanks @serinamarie - all fixed!

**Review by billpalombi — APPROVED** [review]
Great catch, thank you!

**Review by billpalombi — APPROVED** [review]
Great catch, thank you!

**Review by serinamarie — APPROVED** [review]
Thanks @discdiver ! 🙌

**Pull request merged**